source/agora/node/Ledger.d

```diff
@@ -236,6 +249,11 @@ public class Ledger
         if (!this.storage.saveBlock(block))
             assert(0);
 
+        foreach (enrollment; block.header.enrollments)
+            if (!this.enroll_man.updateEnrolledHeight(enrollment.utxo_key,
```

Comment:
This is a great example of what @Geod24 talked about. If `updateEnrolledHeight` fails, we shut down the node immediately with `assert(0);`.

But then the question is, why is `updateEnrolledHeight` returning a boolean status code? It should either throw an Error or just assert(0) itself, and the function should be made `nothrow` and not return any status code.

I will file a separate issue for this, so you don't have to change anything here @TrustHenry.

Reply:
I was thinking the same thing.